src/lib.rs

```diff
@@ -303,6 +303,25 @@ impl Element {
         Attributes::new(&self.buf[self.start..self.end], self.name_end)
     }
 
+    /// Creates a new Element from the given name and attributes.
+    /// attributes are represented as an iterator over (key, value) tuples where
+    /// key and value are both &str.
+    pub fn create<'a, I: Iterator<Item = (&'a str, &'a str)>>(name: &str, attributes: I) -> Element {
```

Comment:
Few remarks:
- you can use `new` instead of `create` and rename the existing (private) `new` function into `from_buffer` for instance (opened for other suggestion)
- `(&'a str, &'a str)` is not symmetric with `Attributes::Item = (&[u8], &str)`. I wanted to avoid utf8 conversion when possible because most of the time we (I ?) compare it with statically known names, thus I could always use `key == b"mykey"`. Having symmetric behavior could help creating Elements based on other Elements (in your test, you had to use `from_utf8`), I think it is a common scenario
  - in your example, we could use `Vec<u8>` and push bytes directly
  - an alternative is to have 2 `Attributes`-like iterators, one pure `&[u8]`, another one `&str`